omniscidb/QueryEngine/ResultSet.cpp

```diff
@@ -1498,9 +1502,34 @@ std::vector<std::pair<const int8_t*, size_t>> ResultSet::getChunkedColumnarBuffe
   std::vector<std::pair<const int8_t*, size_t>> retval;
   retval.reserve(1 + appended_storage_.size());
 
-  retval.emplace_back(
-      storage_->getUnderlyingBuffer() + storage_->getColOffInBytes(column_idx),
-      storage_->binSearchRowCount());
+  auto rows_to_fetch = storage_->binSearchRowCount();
+  if (isTruncated()) {
+    rows_to_fetch = std::min(rows_to_fetch, keep_first_);
+  }
+  retval.emplace_back(storage_->getUnderlyingBuffer() +
```

Comment:
This code is incorrect for cases when `drop_first_ + keep_first_ > storage_->binSearchRowCount()`. It would reference past the storage buffer. Big offsets might force you to skip one or more storages but your code doesn't do it.

Reply:
I think it will be incorrect in case of `drop_first_ >= storage_->binSearchRowCount()`, if only sum is larger I am expecting data form offset to the end of table. (also I am checking it in `OrderedLimitOverSizeSelection` test)

Reply:
The logic here is getting a little hard to understand. Maybe it's better to wrap in some method, not sure what's the best approach here.

Reply:
`storage_` doesn't hold the whole table, so it's legal to drop more rows than `storage_` has.

Reply:
Should be failed with an error or empty selection?

Reply:
BTW, why it's illegal? We simply should skip first storage and use next one.

Reply:
I didn't say it was illegal, I said the opposite. I mentioned missing storage skip in my first comment.